File: app/src/main/java/com/github/batulovandrey/urbandictionarycom/adapter/WordAdapter.java
package com.github.batulovandrey.urbandictionarycom.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.batulovandrey.urbandictionarycom.R;

import java.util.List;

/**
 * @author Andrey Batulov on 27/10/2017
 */

public class WordAdapter extends RecyclerView.Adapter<WordViewHolder> {

    private List<String> mWords;
    private WordClickListener mClickListener;

    public WordAdapter(List<String> words, WordClickListener listener) {
        mWords = words;
        mClickListener = listener;
    }

    @Override
    public WordViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.word_item, null);
        return new WordViewHolder(view, mClickListener);
    }

    @Override
    public void onBindViewHolder(WordViewHolder holder, int position) {
        String word = mWords.get(position);
        holder.mWordTextView.setText(word);
    }

    @Override
    public int getItemCount() {
        return mWords.size();
    }
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/adapter/WordAdapter.kt
package com.github.batulovandrey.unofficialurbandictionary.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.github.batulovandrey.unofficialurbandictionary.R

/**
 * @author Andrey Batulov on 22/12/2017
 */

class WordAdapter(private val mWords: List<String>, private val mClickListener: WordClickListener)
    : RecyclerView.Adapter<WordViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WordViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.word_item, null)
        return WordViewHolder(view, mClickListener)
    }

    override fun onBindViewHolder(holder: WordViewHolder, position: Int) {
        val word = mWords[position]
        holder.setWordText(word)
    }

    override fun getItemCount(): Int {
        return mWords.size
    }

    fun getWord(position: Int) = mWords[position]
}